src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumerableTestExtensions.cs

```diff
@@ -147,6 +147,24 @@ public static void ShouldNotBeEmpty<T>(this IEnumerable<T> actual, [InstantHandl
                 throw new ShouldAssertException(new ExpectedShouldlyMessage(actual, customMessage).ToString());
         }
 
+        public static T ShouldHaveSingleItem<T>(this IEnumerable<T> actual)
+        {
+            return ShouldHaveSingleItem(actual, () => null);
+        }
+
+        public static T ShouldHaveSingleItem<T>(this IEnumerable<T> actual, string customMessage)
+        {
+            return ShouldHaveSingleItem(actual, () => customMessage);
+        }
+
+        public static T ShouldHaveSingleItem<T>(this IEnumerable<T> actual, [InstantHandle] Func<string> customMessage)
+        {
+            if (actual == null || (actual != null && actual.Count() != 1))
```

Comment:
Do we need the second null reference check? Wouldn't  `actual == null || actual.Count() != 1` suffice?